sqlserver/datadog_checks/sqlserver/agent_history.py

```diff
@@ -0,0 +1,137 @@
+import time
+
+from datadog_checks.base import is_affirmative
+from datadog_checks.base.utils.db.utils import DBMAsyncJob, default_json_event_encoding
+from datadog_checks.base.utils.serialization import json
+from datadog_checks.base.utils.tracking import tracked_method
+from datadog_checks.sqlserver.config import SQLServerConfig
+from datadog_checks.sqlserver.const import STATIC_INFO_ENGINE_EDITION, STATIC_INFO_VERSION
+
+try:
+    import datadog_agent
+except ImportError:
+    from ..stubs import datadog_agent
+
+DEFAULT_COLLECTION_INTERVAL = 15
+
+AGENT_HISTORY_QUERY = """\
+SELECT TOP 10
+    j.name,
+    sjh1.job_id,
+    sjh1.step_id,
+    sjh1.step_name,
+    sjh1.instance_id AS step_instance_id,
+    (
+        SELECT MIN(sjh2.instance_id)
+        FROM msdb.dbo.sysjobhistory AS sjh2
+        WHERE sjh2.job_id = sjh1.job_id
+        AND sjh2.step_id = 0
+        AND sjh2.instance_id >= sjh1.instance_id
+    ) AS completion_instance_id,
+    sjh1.run_date,
+    sjh1.run_time,
+    sjh1.run_duration,
+    sjh1.run_status,
+    sjh1.message
+FROM 
+    msdb.dbo.sysjobhistory AS sjh1
+INNER JOIN msdb.dbo.sysjobs AS j
+ON j.job_id = sjh1.job_id
+WHERE 
+    EXISTS (
+        SELECT 1
+        FROM msdb.dbo.sysjobhistory AS sjh2
+        WHERE sjh2.job_id = sjh1.job_id
+        AND sjh2.step_id = 0
+        AND sjh2.instance_id >= sjh1.instance_id{last_instance_id_filter}
+    )
+"""
+def agent_check_getter(self):
+    return self._check
+
+class SqlserverAgentHistory(DBMAsyncJob):
+    def __init__(self, check, config: SQLServerConfig):
+        # do not emit any dd.internal metrics for DBM specific check code
+        self.tags = [t for t in check.tags if not t.startswith('dd.internal')]
+        self.log = check.log
+        self._config = config
+        collection_interval = float(
+            self._config.agent_jobs_interval
+        )
+        if collection_interval <= 0:
+            collection_interval = DEFAULT_COLLECTION_INTERVAL
+        self.collection_interval = collection_interval
+        self._last_history_id = check._last_history_id
```

Comment:
why does this field need to be set in the main check?

Reply:
For some reason I thought the main check reinitialized the class every time it ran. Will change this.